experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1521,6 +1521,20 @@ def _validate_required_or_excluded_experiments(self, value):
 
         return value
 
+    def add_to_versioned_errs(
+        self,
+        errs: list[str],
+        version: NimbusFeatureVersion,
+        existing_errs_with_version: dict[str, list[NimbusFeatureVersion]],
+    ) -> dict[str, list[NimbusFeatureVersion]]:
+        errs_with_version = existing_errs_with_version
+        for e in errs:
+            if e in errs_with_version:
+                errs_with_version[e].append(version)
+            else:
+                errs_with_version[e] = [version]
+        return errs_with_version
+
```

Comment:
Python has something for this exact use case: [`collections.defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict). It is a dict subclass that accepts a factory that will be used to generate values for keys not present in the dict. It allows you to do:

```python
>>> from collections import defaultdict
>>> xs = defaultdict(list)
>>> xs[0].append("a")
>>> xs[1].append("b")
>>> xs
defaultdict(<class 'list'>, {0: ['a'], 1: ['b']})
```

Additionally, you can also use [`list.setdefault`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault) if you only need that behaviour sometimes:

```python
>>> ys = {}
>>> ys.setdefault(0, []).append("a")
>>> ys
{0: ['a']}
```

I would reccomend using `defaultdict` in this instance and then return it with `dict(xs)` where xs is the defaultdict instance. That way the returned value will not have the defaultdict behaviour.